substrate/frame/identity/src/lib.rs

```diff
@@ -1164,116 +1219,184 @@ pub mod pallet {
+					if let Some(authority_account) =
+						AuthorityOf::<T>::get(&suffix).map(|auth_info| auth_info.account_id)
+					{
+						let err_amount =
+							T::Currency::unreserve(&authority_account, username_deposit);
+						debug_assert!(err_amount.is_zero());
+					}
+				},
+				Provider::Allocation => {
+					// We don't refund the allocation, it is lost.
+				},
+				Provider::System => return Err(Error::<T>::InvalidTarget.into()),
+			}
+			Self::deposit_event(Event::DanglingUsernameRemoved {
+				who: username_info.owner.clone(),
+				username,
+			});
 			Ok(Pays::No.into())
 		}
-	}
-}
-
-impl<T: Config> Pallet<T> {
-	/// Information that is pertinent to identify the entity behind an account. First item is the
-	/// registration, second is the account's primary username.
-	///
-	/// TWOX-NOTE: OK ― `AccountId` is a secure hash.
-	pub fn identity(
-		who: T::AccountId,
-	) -> Option<(
-		Registration<BalanceOf<T>, T::MaxRegistrars, T::IdentityInformation>,
-		Option<Username<T>>,
-	)> {
-		IdentityOf::<T>::get(who)
-	}
-
-	/// The super-identity of an alternative "sub" identity together with its name, within that
-	/// context. If the account is not some other account's sub-identity, then just `None`.
-	pub fn super_of(who: T::AccountId) -> Option<(T::AccountId, Data)> {
-		SuperOf::<T>::get(who)
-	}
-
-	/// Alternative "sub" identities of this account.
-	///
-	/// The first item is the deposit, the second is a vector of the accounts.
-	///
-	/// TWOX-NOTE: OK ― `AccountId` is a secure hash.
-	pub fn subs_of(
-		who: T::AccountId,
-	) -> (BalanceOf<T>, BoundedVec<T::AccountId, T::MaxSubAccounts>) {
-		SubsOf::<T>::get(who)
-	}
-
-	/// The set of registrars. Not expected to get very big as can only be added through a
-	/// special origin (likely a council motion).
-	///
-	/// The index into this can be cast to `RegistrarIndex` to get a valid value.
-	pub fn registrars() -> BoundedVec<
-		Option<
-			RegistrarInfo<
-				BalanceOf<T>,
-				T::AccountId,
-				<T::IdentityInformation as IdentityInformationProvider>::FieldsIdentifier,
-			>,
-		>,
-		T::MaxRegistrars,
-	> {
-		Registrars::<T>::get()
-	}
 
-	/// A map of the accounts who are authorized to grant usernames.
-	pub fn authority(who: T::AccountId) -> Option<AuthorityPropertiesOf<T>> {
-		UsernameAuthorities::<T>::get(who)
-	}
-
-	/// Reverse lookup from `username` to the `AccountId` that has registered it. The value should
-	/// be a key in the `IdentityOf` map, but it may not if the user has cleared their identity.
-	///
-	/// Multiple usernames may map to the same `AccountId`, but `IdentityOf` will only map to one
-	/// primary username.
-	pub fn username(username: Username<T>) -> Option<T::AccountId> {
-		AccountOfUsername::<T>::get(username)
-	}
+		/// Start the process of removing a username by placing it in the unbinding usernames map.
+		/// Once the grace period has passed, the username can be permanently deleted by calling
```

Comment:
It can be reissued, even with the same account ID if both the user and authority decide to do so. I guess the word "permanently" was misleading there, so I removed it.